src/parakeet/Driver.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+	Copyright 2021 OpenJAUS, LLC (dba MechaSpin). Subject to the MIT license.
+*/
+
+#include <parakeet/Driver.h>
+#include <parakeet/exceptions/NotConnectedToSensorException.h>
+
+#include <iostream>
+
+namespace mechaspin
+{
+namespace parakeet
+{
+	void Driver::stop()
+	{
+        runUpdateThread = false;
+
+        if(updateThread.joinable())
+        {
+            updateThread.join();
+        }
+	}
+
+	void Driver::close()
+	{
+        bool isc = isConnected();
+        bool runUpdate = runUpdateThread;
+        bool isJoin = updateThread.joinable();
+        if (isConnected() || runUpdateThread || updateThread.joinable())
+        {
+           stop();
+        }
+	}
+
+    void Driver::start()
+    {
+        throwExceptionIfNotConnected();
+
+        runUpdateThread = true;
+        updateThreadStartTime = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch());
+        updateThreadFrameCount = 0;
+        updateThread = std::thread([&] { this->updateThreadMainLoop(); });
+    }
+
+	void Driver::updateThreadMainLoop()
+	{
+        while (runUpdateThread)
+        {
+			if(updateThreadCallbackFunction)
+			{
+				updateThreadCallbackFunction();
+			}
+		}
+	}
+
+	void Driver::throwExceptionIfNotConnected()
```

Comment:
Consider renaming this to something assertIsConnected() just to shorten up the method name. I feel like the word assery as pretty clear meaning; this thing MUST be true otherwise there's an issue.